cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/schedule/SchedulerServiceTest.java

```diff
@@ -166,6 +169,36 @@ public void testSimpleSchedulerLifecycle() throws Exception {
     checkState(Scheduler.ScheduleState.NOT_FOUND, scheduleIds);
   }
 
+  @Test
+  public void testScheduleUpdate() throws Exception {
+    AppFabricTestHelper.deployApplication(namespace, AppWithWorkflow.class);
+    ApplicationSpecification applicationSpecification = store.getApplication(appId);
+
+    schedulerService.schedule(program, programType, ImmutableList.of(timeSchedule1));
+    applicationSpecification = createNewSpecification(applicationSpecification, program, programType, timeSchedule1);
+    store.addApplication(appId, applicationSpecification, locationFactory.create("app"));
+    List<String> scheduleIds = schedulerService.getScheduleIds(program, programType);
+    Assert.assertEquals(1, scheduleIds.size());
+    checkState(Scheduler.ScheduleState.SUSPENDED, scheduleIds);
+
+    schedulerService.updateSchedule(program, programType, updatedTimeSchedule1);
+
+    schedulerService.resumeSchedule(program, programType, "Schedule1");
+    checkState(Scheduler.ScheduleState.SCHEDULED, scheduleIds);
+
+    List<ScheduledRuntime> scheduledRuntimes = schedulerService.nextScheduledRuntime(program, programType);
```

Comment:
It would be a good idea to check the `nextScheduledRuntime` once before to assert the first schedule.